### PR TITLE
EPMDJ-8705: Select agent panel

### PR DESCRIPTION
**Pull request by IliaSolovev**

### CODE REVIEW

File: src/ui/components/panels/select-agent-panel.tsx [hunk 1]
```diff
@@ -13,12 +13,163 @@
...
+  const setPanel = useSetPanelContext();
+  const isPreregisteredAgent = agentType === "Java" && !agentVersion;
+  const isRegistering = status === AGENT_STATUS.REGISTERING;
+  const isSelectedAgent = agentId === id;
+  return (
+    <Row
+      selected={isSelectedAgent}
+      isGroupAgent={Boolean(group)}
+      isRegistering={isRegistering}
+      isPreregisteredAgent={isPreregisteredAgent}
```
Comment (RomanAchilov) [inline src/ui/components/panels/select-agent-panel.tsx:75]: Can we decompose these props?

File: src/ui/components/panels/select-agent-panel.tsx [hunk 2]
```diff
@@ -13,12 +13,163 @@
...
+          name: "agentDashboard",
+          params: { agentId: id, buildVersion },
+        }));
+        setPanel(null);
+      }}
+    >
+      <Badge color="green" bold tw="opacity-0">NEW</Badge>
+      {isRegistering
+        ? <div tw="flex justify-center items-center"><Spinner /></div>
+        : <CubeWrapper tw="ml-2" isActive={isSelectedAgent} id="Agent 4">{convertAgentName(name)}</CubeWrapper>}
```
Comment (RomanAchilov) [inline src/ui/components/panels/select-agent-panel.tsx:87]: id="Agent 4"

?

File: src/ui/components/panels/select-agent-panel.tsx [hunk 3]
```diff
@@ -13,12 +13,163 @@
...
+
+const GroupRow = ({ agents = [], group: { id = "", name: groupName = "", description } }: GroupRowProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { groupId } = useAgentRouteParams();
+  const { push } = useHistory();
+  const setPanel = useSetPanelContext();
+  const isSelectedGroup = groupId === id;
+
+  return (
+    <div tw="rounded-lg bg-[#2B292B]">
```
Comment (RomanAchilov) [inline src/ui/components/panels/select-agent-panel.tsx:113]: add to tailwind.config

File: src/ui/modules/system-setting-step/system-settings-step.tsx [hunk 4]
```diff
@@ -31,27 +31,7 @@ export const SystemSettingsStep = () => (
...
-                        &quot;/&quot; in a package path.
-                      </div>
-                    </div>
-                  )}
-                >
-                  <Icons.Info />
-                </Tooltip>
-              </div>
-            )}
+            label="asd"
```
Comment (RomanAchilov) [inline src/ui/modules/system-setting-step/system-settings-step.tsx:34]: ?

File: src/ui/components/panels/select-agent-panel.tsx [hunk 5]
```diff
@@ -13,12 +13,163 @@
...
+  const agents = agentsList.filter((agent) => !agent.group);
+  const groupsAgents = agentsList.filter((agent) => agent.group);
+  const groups = groupsList.map((group) => ({
+    group,
+    agents: groupsAgents.filter((agent) => group.name === agent.group),
+  }));
+
+  return (
+    <Panel header={<div tw="flex items-center h-21">Select Agent</div>} isOpen={isOpen} onClosePanel={onClosePanel}>
+      <div tw="w-[1024px] text-monochrome-medium-tint text-14 leading-20">
```
[the remark this answers is not in the document]
Comment (IliaSolovev) [reply src/ui/components/panels/select-agent-panel.tsx:44]: Value from design))

File: src/ui/components/panels/select-agent-panel.tsx (hunk 1 above)
Comment (IliaSolovev) [reply src/ui/components/panels/select-agent-panel.tsx:75]: Now I don`t know how decompose it but I hope that BE will send "registration" and "preregistered" status.And i will pass it here like a single props

File: src/ui/components/panels/select-agent-panel.tsx [hunk 6]
```diff
@@ -13,12 +13,163 @@
...
 import { PanelProps } from "./panel-props";
+import { Cube } from "../cubes";
+import { useSetPanelContext } from "../navigation";
+
+export const SelectAgentPanel = ({ isOpen, onClosePanel }: PanelProps) => {
+  const agentsList = useAdminConnection<Agent[]>("/api/agents") || [];
+  const groupsList = useAdminConnection<ServiceGroup[]>("/api/groups") || [];
+  const agents = agentsList.filter((agent) => !agent.group);
+  const groupsAgents = agentsList.filter((agent) => agent.group);
+  const groups = groupsList.map((group) => ({
```
[the remark this answers is not in the document]
Comment (IliaSolovev) [reply src/ui/components/panels/select-agent-panel.tsx:37]: but it had 3 spread operators

File: src/ui/components/panels/select-agent-panel.tsx [hunk 7]
```diff
@@ -13,12 +13,163 @@
...
+    group,
```
[the remark this answers is not in the document]
Comment (IliaSolovev) [reply src/ui/components/panels/select-agent-panel.tsx:38]: yes

**Pull request merged**